### PR TITLE
Sabbir task/api gateway and dynamodb

### PR DESCRIPTION
**Pull request by sabbir80**
Fixes #

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [ ] Improvement
- [x] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?
Introduced Api gateway and a new DynamoDb table CustomerUserActivityTable.
## What was done
Added post method and get method for CustomerUserActivityTable 
## How to test

- [ ] Needs regression testing (add QA needed label)

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

## Reminder list

- Determine whether REST API Documentation should be updated
- Make an assessment on whether unit tests should be updated and/or added
- Ensure follow up tasks are linked to this PR

QA - if regression testing was requested:

- [ ] QA has verified

### CODE REVIEW

File: firstLambda/handler.py [hunk 1]
```diff
@@ -46,6 +48,245 @@ def is_video_file(filename):
...
+def generate_gsisk(data):
+    key = {}
+    if data.get('customer') and data.get('initialDate'):
+        gsisk = f'{data.get("customer")}#{data.get("initialDate")}'
+        key['from_gsisk'] = gsisk
+        if data.get('finalDate'):
+            com_gsisk = f'{data.get("customer")}#{data.get("finalDate")}'
+            key['to_gsisk'] = com_gsisk
+    return key
+
```
Comment (mhazad) [inline firstLambda/handler.py:83]: The utility methods for generating the keys should be plain and simple and they should not do a validation of the data object as its not its responsibility. 

generate_pk or get_pk_for_user_activity method should the customer as its parameter and should return 
```python
return f'cus#{data.get("customer")}'
```

and the same goes for other methods that will only take relevant parameters to generate the keys.

**Review by mhazad — CHANGES_REQUESTED** [review]
Good progress! Please address the feedback.

File: firstLambda/handler.py (hunk 1 above)
Comment (sabbir80) [reply firstLambda/handler.py]: Addressed the feedback in 89110f0

**Review by mhazad — CHANGES_REQUESTED** [review]
LGTM

**Pull request merged**